### PR TITLE
docs(input): add example of a clearable input

### LINKED ISSUE #3725: mdInput clear input button
#### Bug, feature request, or proposal:
Feature Request

#### What is the expected behavior?
When an mdInput text field has a value, there should be a clear (X) button on the right of the input to clear the text field.

#### What is the current behavior?
No button, user has to manually clear, or dev has to create another directive

#### What is the use-case or motivation for changing an existing behavior?
I'm currently working on a search field, and expected the `<input type="search" mdinput></input>` to give users a clear button, as per the docs here https://material.io/guidelines/components/text-fields.html#text-fields-search-filter - however I'm either missing something, or it's not provided.

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #3725

### CODE REVIEW

**Comment by jelbourn** [comment]
I'm hesitant to change the examples here until we have them building.

**Comment by jelbourn** [comment]
@mmalerba can you rebase?

**Comment by mmalerba** [comment]
done

File: src/examples/input-clearable/input-clearable-example.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<md-input-container class="example-input-container">
+  <input mdInput type="text" placeholder="Clearable input" [(ngModel)]="value"/>
+  <md-icon mdSuffix *ngIf="value" (click)="value=''">close</md-icon>
```
Comment (jelbourn) [inline src/examples/input-clearable/input-clearable-example.html]: Should be an `md-icon-button` for a11y

Comment (mmalerba) [inline src/examples/input-clearable/input-clearable-example.html]: So actually... `md-icon-button` doesn't really work well as a suffix currently. It's too tall and it causes the placeholder and input text to be misaligned. Filed #4382 to fix this

**Comment by mmalerba** [comment]
icon-buttons are supported inside inputs now, updated example

**Comment by jelbourn** [comment]
Needs rebase

**Pull request merged**